### DISCUSSION TITLE
Automatic sync between property lists and property search form

### DISCUSSION THREAD

**Discussion by fredericalpers** [onOffice-Web-Org/oo-wp-plugin#736] [General]
While discussing https://github.com/onOffice-Web-Org/oo-wp-plugin/issues/613 the question arose as to whether it would be possible to have an automatic sync between a property list (default.php) and a property list as a search form (SearchForm.php)?

@yeneastgate @dai-eastgate do you think this could be possible? If so do you have any suggestions on how to achive this the best way possible?

**Comment by dai-eastgate** [comment]
@fredericalpers Yes, we will investigate and let you know as soon as possible.

**Reply by dai-eastgate** [reply]
@fredericalpers by the way, do I need to prioritize investigating this issue first?

**Reply by fredericalpers** [reply]
You do not need to prioritize this investigation :)

**Comment by fredericalpers** [comment]
@dai-eastgate after finishing up the open issues in v4.21, please go ahead and investigate this before starting v4.22, thank you :)

**Reply by dai-eastgate** [reply]
@fredericalpers  I got it

**Comment by dai-eastgate** [comment]
@fredericalpers  I have a solution to resolve "automatic sync between property lists and property search form" , Please watch the demo video:

https://files.fm/u/xaapqbrg27#/view/rw574p7h2y

=> If you agree with this solution, I will  implement issue #613

**Comment by fredericalpers** [comment]
@dai-eastgate sorry I think this was poorly worded by me. We specifically want to know if we can sync this in the plugin backend, so we do not have to go through the following:

Currently you have to set up one property list with the template searchform and will have to select all the fields as "show in search" and then you will have to set up a second list where you also will have to select all fields as "show in search", otherwise the searchform does not work.

**Comment by dai-eastgate** [comment]
@fredericalpers The "list config fields" of a listview with "search Form" template will be synchronized with the "list config fields" of a listview with “default” template (which will be selected in the page forwarding page that you mentioned in #613)?

![image](https://github.com/onOffice-Web-Org/oo-wp-plugin/assets/106214469/b31708c8-3033-4d2f-bcbf-f26547d1e557)

The issue you are facing is when only using the search form (searchform.php) on the homepage, is it correct? as in the video:

https://github.com/onOffice-Web-Org/oo-wp-plugin/assets/106214469/917373b9-3114-4c5a-b885-52bfb840b708

Note: If needed we can call to talk about at daily meeting in tomorrow

**Comment by dai-eastgate** [comment]
> we do not have to go through the following:
> 
> Currently you have to set up one property list with the template searchform and will have to select all the fields as "show in search" and then you will have to set up a second list where you also will have to select all fields as "show in search", otherwise the searchform does not work.

@fredericalpers we have tried implementing a solution to resolve your problem.
we automatically sync "all fields that have been added to the field list" of a property list as a search form (SearchForm.php) to "field list" of a property list (default.php).

https://files.fm/u/rvtkrskf2m#/view/qrrauh8xtg
Pls, watch it and respond your opinion and confirm, is this your Desired state?
You can review the code and test it at "45019-property-search-and-form-action-redirect" branch

**Comment by fredericalpers** [comment]
@dai-eastgate We will review this carefully and give you feedback as soon as possible, thank you
